lib/gollum-lib/file.rb

```diff
@@ -19,13 +19,15 @@ def initialize(wiki)
     # Public: The url path required to reach this page within the repo.
     #
     # Returns the String url_path
+
     def url_path
-      path = self.path
-      if path.include?('/')
-        path = path.sub(/\/[^\/]+$/, '/')
-        path << name
-      end
-      path
+      path = if self.path.include?('/')
+        self.path.sub(/\/[^\/]+$/, '/')
+          else
+            ''
+          end
+      path = path[@wiki.page_file_dir.length+1..-1] if @wiki.page_file_dir # Chop off the page file dir plus the first slash if necessary 
+      path << name
```

Comment:
Not sure the `<< name` here is appropriate, as it looks like the `name` method sometimes returns the full `path` itself! If it's a mistake, this was already broken. Help appreciated.

Reply:
Given this issue and substantial overlap, I think we should consider making `File` a subclass of `Page`, or having some other class hierarchy that avoids duplication.

Reply:
> having some other class hierarchy that avoids duplication.

Yes, that makes sense. I would think that `Gollum::Page` would be a special case of the more general `Gollum::File` class, but that may just be a personal preference.

Reply:
> Not sure the `<< name` here is appropriate

This was introduced [here](https://github.com/gollum/gollum-lib/commit/b10a0c98ba12a5ed8f8a05ff69d5ca68f1b65ad9) to let `File.url_path` behave in the same way as `Page.url_path`. Another reason for creating a superclass I would say.

Reply:
> as it looks like the `name` method sometimes returns the full `path` itself

Only in the sense that the "full path" sometimes just _is_ the name of the page (any page at root level).